xarray/core/indexing.py

```diff
@@ -875,10 +875,10 @@ def as_indexable(array):
         return PandasIndexingAdapter(array)
     if is_duck_dask_array(array):
         return DaskIndexingAdapter(array)
-    if hasattr(array, "__array_function__"):
-        return NdArrayLikeIndexingAdapter(array)
     if hasattr(array, "__array_namespace__"):
         return ArrayApiIndexingAdapter(array)
+    if hasattr(array, "__array_function__"):
```

Comment:
I'm confused now, why does the indexing adapter affect what's used for reductions?

Reply:
I don't think that's the case. What I mentioned in the meeting was that we could use `__array_function__` as a fallback for `nan*` reductions, whereas choosing the wrong indexing adapter will cause `xarray` to try indexing with a method the array type might not support.

I guess this is kinda hard to figure out implicitly, and I'd love to have a way (a special attribute like `__xarray_preferred_array_protocol__`?) to explicitly communicate which protocol is preferred. That way, we won't have to worry about `cupy` and `pint` implementing the Array API, and `cubed` can just set that attribute to state that it considers `__array_function__` a fallback and instead prefers to use `__array_namespace__` indexing.

Reply:
> __array_function__ as a fallback for nan* reductions, 

:+1: sorry I got confused.